src/app/beer_garden/api/stomp/server.py

```diff
@@ -128,42 +128,43 @@ def __init__(
-                try:
-                    self.conn.connect(
-                        username=self.username,
-                        passcode=self.password,
-                        wait=True,
-                        headers={"client-id": self.username},
-                    )
-                    if self.subscribe_destination:
-                        self.conn.subscribe(
-                            destination=self.subscribe_destination,
-                            id=self.username,
-                            ack="auto",
-                            headers={
-                                "subscription-type": "MULTICAST",
-                                "durable-subscription-name": self.subscribe_destination,
-                            },
+        if self.host_and_ports:
+            if (
+                self.host_and_ports[0][0]
+                and self.host_and_ports[0][1]
+                and self.subscribe_destination
+            ):
+                while not self.conn.is_connected() and self.bg_active:
+                    try:
+                        self.conn.connect(
+                            username=self.username,
+                            passcode=self.password,
+                            wait=True,
+                            headers={"client-id": self.username},
                         )
-                    if connected_message is not None and self.conn.is_connected():
-                        logger.info("Stomp successfully " + connected_message)
-
-                except Exception as e:
-                    logger.warning(str(e))
-                    logger.warning(
-                        f"Affected gardens are {[garden.get('name') for garden in gardens]}"
-                    )
-                    logger.warning(
-                        "Waiting %.1f seconds before next attempt", wait_time
-                    )
-                    time.sleep(wait_time)
-                    wait_time = min(wait_time * 2, 30)
+                        if self.subscribe_destination:
+                            self.conn.subscribe(
+                                destination=self.subscribe_destination,
+                                id=self.username,
+                                ack="auto",
+                                headers={
+                                    "subscription-type": "MULTICAST",
+                                    "durable-subscription-name": self.subscribe_destination,
+                                },
+                            )
+                        if connected_message is not None and self.conn.is_connected():
+                            logger.info("Stomp successfully " + connected_message)
+
+                    except Exception as e:
+                        logger.warning(str(e))
+                        logger.warning(
+                            f"Affected gardens are {[garden.get('name') for garden in gardens]}"
```

Comment:
If you separate these into 2 calls there could be any number of lines in between them in the actual log file. It's safer to combine them into one call:

```python
logger.warning("Error connecting: {e}. Affected gardens are {[garden.get('name') for garden in gardens]}")
```